plugins/book-data-extension/src/plugin/index.ts

```diff
@@ -10,6 +10,8 @@ export const createBookDataExtensionPlugin = (): BookFetcherPlugin => {
 		const extensionAvailable = await listenForExtension(`BOOK_FETCHER:PONG`, 500);
 
 		if (!extensionAvailable) return [];
+		console.log("Extension Available");
+		localStorage.setItem("extensionAvailable", "true");
```

Comment:
I don't think localstorage is conceptually the right spot to store this info. I think a svelte store would be a better fit.

Also, the check for the extension presence should happen on page load IMO, not at the first attempt to invoke `fetchBookData`.

Reply:
@silviot Yeah, I agree that checking for the extension should happen on page load because the way I implemented it would display a disabled button when the app loads for the first time (before any call to fetchBookData).

My reasoning behind using localStorage is 
a) I couldn't use a svelte store because the plugin package does not use svelte. I could use a different way to manage the state, this way was just the most straight forward.
b) Despite localStorage not being reactive by design, the fetchBookData call listens for the extension availability (and updates the localStorage) on each call so the value would never go stale and it would be constantly synced.

Reply:
@fadwamahmoud Can we use a store-like object that can then be imported from the rest of the app?

Something like this [ChatGPT produced this code and I haven't thoroughly checked it]:

```typescript
class BooleanStore {
  private _value: boolean;
  private _subscribers: Array<(value: boolean) => void> = [];

  constructor(initialValue: boolean) {
    this._value = initialValue;
  }

  subscribe(run: (value: boolean) => void): () => void {
    this._subscribers.push(run);
    run(this._value); // Run the subscriber immediately with the current value

    return () => {
      this._subscribers = this._subscribers.filter((sub) => sub !== run);
    };
  }

  set(value: boolean): void {
    this._value = value;
    this._subscribers.forEach((sub) => sub(value));
  }

  update(updater: (currentValue: boolean) => boolean): void {
    this.set(updater(this._value));
  }
}

// Usage:
// const store = new BooleanStore(true);
// store.subscribe(value => console.log(value));
// store.set(false);
// store.update(value => !value);
```

Since there's no load funciton here, the check (and storing the result of the check into the store like object) could then happen "at import time" (or "at module level") in python parlance.

"At the top level of the module"
"Outside of any component or function"
"In the global scope of the module"